ironfish-rust/src/transaction/mod.rs

```diff
@@ -423,20 +431,19 @@ impl ProposedTransaction {
             binding_verification_key += mint.value_commitment_point();
         }
 
-        for burn in &self.burns {
-            binding_signature_key -= burn.value_commitment.randomness;
-            binding_verification_key -= burn.value_commitment_point();
-        }
-
         let private_key = PrivateKey(binding_signature_key);
         let public_key =
             PublicKey::from_private(&private_key, VALUE_COMMITMENT_RANDOMNESS_GENERATOR);
 
-        check_value_consistency(
-            &public_key,
-            &binding_verification_key,
-            *self.value_balances.fee(),
-        )?;
+        let value_balance =
+            calculate_value_balance(&binding_verification_key, *self.value_balances.fee(), burns)?;
+
+        // Confirm that the public key derived from the binding signature key matches
+        // the final value balance point. The binding verification key is how verifiers
+        // check the consistency of the values in a transaction.
+        if value_balance != public_key.0 {
```

Comment:
Moved this check out of `check_value_consistency`, so that we could re-use the function in both the proposed transaction and the final transaction to de-dupe a bit of logic